### PR TITLE
fix: PostPolicyV4 classes could be improved

### PR DESCRIPTION
**Pull request by dmitry-fa**

### CODE REVIEW

File: google-cloud-storage/src/main/java/com/google/cloud/storage/PostPolicyV4.java [hunk 1]
```diff
@@ -18,56 +18,123 @@
...
+import java.util.List;
 import java.util.Map;
 import java.util.Objects;
 import java.util.Set;
+import java.util.concurrent.TimeUnit;
 
 /**
- * Presigned V4 post policy.
+ * Presigned V4 post policy. Instances of {@code PostPolicyV4} include a URL and a map of fields
+ * that can be specified in an HTML form to submit a POST request to upload an object.
```
Comment (athakor) [inline google-cloud-storage/src/main/java/com/google/cloud/storage/PostPolicyV4.java:36]: in an HTML -> in HTML

File: google-cloud-storage/src/main/java/com/google/cloud/storage/PostPolicyV4.java [hunk 2]
```diff
@@ -18,56 +18,123 @@
...
     private PostFieldsV4(Map<String, String> fields) {
-      this.fieldsMap = fields;
+      validateFields(fields);
+      this.fieldsMap = Collections.unmodifiableMap(fields);
     }
 
+    /**
+     * Constructs {@code PostPolicyV4.PostFieldsV4} object of the given field map.
+     *
+     * @param fields a map
```
Comment (athakor) [inline google-cloud-storage/src/main/java/com/google/cloud/storage/PostPolicyV4.java]: a map -> a map of valid fields

File: google-cloud-storage/src/main/java/com/google/cloud/storage/PostPolicyV4.java [hunk 3]
```diff
@@ -112,8 +179,9 @@ public Builder setContentEncoding(String contentEncoding) {
         return this;
       }
 
+      /** @deprecated Invocation this method has no effect */
+      @Deprecated
```
Comment (athakor) [inline google-cloud-storage/src/main/java/com/google/cloud/storage/PostPolicyV4.java:188]: if this method has no effect then why you kept it here? if you deprecate it then we must have an alternate method of this.

File: google-cloud-storage/src/main/java/com/google/cloud/storage/PostPolicyV4.java (hunk 1 above)
Comment (dmitry-fa) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/PostPolicyV4.java:36]: 'in HTML', but 'in an HTML form'

File: google-cloud-storage/src/main/java/com/google/cloud/storage/PostPolicyV4.java (hunk 2 above)
Comment (dmitry-fa) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/PostPolicyV4.java]: done

File: google-cloud-storage/src/main/java/com/google/cloud/storage/PostPolicyV4.java [hunk 4]
```diff
@@ -131,9 +131,9 @@ private PostFieldsV4(Map<String, String> fields) {
     /**
      * Constructs {@code PostPolicyV4.PostFieldsV4} object of the given field map.
      *
-     * @param fields a map
+     * @param fields a map the HTML form fields
```
Comment (athakor) [inline google-cloud-storage/src/main/java/com/google/cloud/storage/PostPolicyV4.java]: nit: a map the HTML form fields -> a map of the HTML form fields

File: google-cloud-storage/src/main/java/com/google/cloud/storage/PostPolicyV4.java (hunk 1 above)
Comment (dmitry-fa) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/PostPolicyV4.java:36]: done

File: google-cloud-storage/src/main/java/com/google/cloud/storage/PostPolicyV4.java (hunk 3 above)
Comment (dmitry-fa) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/PostPolicyV4.java:188]: added an explanation of why this method makes no effect.

File: google-cloud-storage/src/main/java/com/google/cloud/storage/PostPolicyV4.java [hunk 5]
```diff
@@ -179,7 +179,12 @@ public Builder setContentEncoding(String contentEncoding) {
         return this;
       }
 
-      /** @deprecated Invocation of this method has no effect. */
+      /**
+       * @deprecated Invocation of this method has no effect, because All valid HTML form fields
```
Comment (athakor) [inline google-cloud-storage/src/main/java/com/google/cloud/storage/PostPolicyV4.java]: looks good to me with final 
nit: i guess `All` should be  in lowercase `all`

**Review by athakor — APPROVED** [review]

File: google-cloud-storage/src/main/java/com/google/cloud/storage/PostPolicyV4.java (hunk 5 above)
Comment (dmitry-fa) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/PostPolicyV4.java]: done

**Pull request closed without merging**